### ISSUE TITLE
camera.controls.setOrbitPoint() breaks SimpleRaycaster

### ISSUE THREAD

**Issue by CabeDigicorp** [ThatOpen/engine_components#512]
### Describe the bug 📝

If I change the camera orbit point using Yomotsu’s camera.controls.setOrbitPoint() method, SimpleRaycaster casts in the wrong direction and consequently Highlighter goes into chaos.

Here is a small video: initially some correct highlighting, then setOrbitpoint() is called (orbit point marked by greenish sphere), after that highlighting is effectively random.
[Registrazione dello schermo 2024-10-15 151500.zip](https://github.com/user-attachments/files/17390649/Registrazione.dello.schermo.2024-10-15.151500.zip)

### System Info 💻

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 AMD Ryzen 9 6900HX with Radeon Graphics
    Memory: 17.10 GB / 29.74 GB
  Binaries:
    Node: 22.1.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.12.1 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (129.0.2792.89)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/ThatOpen/engine_components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a repository issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by yaras-phoenix** [comment]
This seems to be resolved in camera-controls v2.9.0. Could you update the library in pakcage.json?

**Comment by agviegas** [comment]
In the latest version of the [highlighter example](https://docs.thatopen.com/Tutorials/Components/Front/Highlighter): 

```ts
window.addEventListener("keydown", (e) => {
  if (e.code === "KeyP") {
    alert("Key P pressed!");
    world.camera.controls.setOrbitPoint(1, 3, 5);
  }
});
```

The result:


https://github.com/user-attachments/assets/26fcc59e-88ab-4ec8-befd-acb30d0ff0dc

Full code here:

[example.zip](https://github.com/user-attachments/files/17480084/example.zip)

**Comment by RyugaRyuzaki** [comment]
> In the [latest version](https://www.npmjs.com/package/@thatopen/components-front?activeTab=versions) of the [highlighter example](https://docs.thatopen.com/Tutorials/Components/Front/Highlighter):
> 
> ```ts
> window.addEventListener("keydown", (e) => {
>   if (e.code === "KeyP") {
>     alert("Key P pressed!");
>     world.camera.controls.setOrbitPoint(1, 3, 5);
>   }
> });
> ```
> 
> The result:
> 
>  2024-10-22.19-30-11.mp4 
> Full code here:
> 
> [example.zip](https://github.com/user-attachments/files/17480084/example.zip)
> 
> Closing this for now as it seems `setOrbitPoint` doesn't break the raycaster, but if you can provide an example where we can reproduce it, I'll reopen this. Let me know how it goes!

It appears when we : 
```
let count=0
window.addEventListener("keydown", (e) => {
  if (e.code === "KeyP") {
    alert("Key P pressed!");
    world.camera.controls.setOrbitPoint(count*1, count*3, count*5);
    count++
  }
});
```
I mean we change the point.

**Comment by RyugaRyuzaki** [comment]
It's strange.
In this [function](https://github.com/ThatOpen/engine_components/blob/main/packages/front/src/fragments/Highlighter/index.ts#L642)  i call : 
```
const casters = this.components.get(OBC.Raycasters);
    const intersection = caster.castRay( allMeshes );
    if ( intersection ) {
      const {x,y,z}=intersection.point
      this.controls.setOrbitPoint(x,y,z)
    }
```
But when in [this function](https://github.com/ThatOpen/engine_components/blob/main/packages/front/src/fragments/Highlighter/index.ts#L663)  it work again ( i mean select is correct , the event "controlEnd" of cameraControl end)

**Comment by agviegas** [comment]
@RyugaRyuzaki using this:

```ts
let count=0
window.addEventListener("keydown", (e) => {
  if (e.code === "KeyP") {
    alert("Key P pressed!");
    world.camera.controls.setOrbitPoint(count*1, count*3, count*5);
    count++
  }
});
```
Same result (works as expected):

https://github.com/user-attachments/assets/420c748a-af33-4575-8df7-53174241a472

**Comment by RyugaRyuzaki** [comment]
> @RyugaRyuzaki using this:
> 
> ```ts
> let count=0
> window.addEventListener("keydown", (e) => {
>   if (e.code === "KeyP") {
>     alert("Key P pressed!");
>     world.camera.controls.setOrbitPoint(count*1, count*3, count*5);
>     count++
>   }
> });
> ```
> 
> Same result (works as expected):
> 
>  2024-10-23.13-21-17.mp4

After you setOrbitPoint, can you rotate camera, then raycaster again?

**Comment by agviegas** [comment]
@CabeDigicorp can you reproduce this with the simple [worlds example](https://github.com/ThatOpen/engine_components/blob/main/packages/core/src/core/Worlds/example.ts)? 🤔

**Comment by CabeDigicorp** [comment]
@agviegas On Friday I prepared an example by "stripping down" our app, keeping only the relevant stuff. But, good news! the bug suddenly disappeared.
So I have just been able to track down the odd behaviour and identify the actual culprit: there was a well-hidden mistake of mine in enabling/disabling the renderer - which I need to do to to keep CPU usage at reasonable level.
I am waiting for a final confirmation from our test team, but I'm confident that no further action will be needed on your side. Sorry for the false alarm and thank you for your help!
